### PR TITLE
Tail and Head commands reimplementation with an helper function and a new error type. + Updating documentation (mans files and Manual.txt file).

### PR DESCRIPTION
**Pull request by mdaadoun**
* A new error call added : **Missing arguments**
* Replaced hardcoded error from **pexecc**, **fpexecc** and **mann** with this new error call.
* Reimplementation of the **head** command while adding the **tail** command. Because they use most of the same logic, created a subcommand **headtail** to use by both of them.
* Improved the **documentation** (in comments and man files) for errors, commands and the new helper function, future with generation of manual from python docstring ? ([PEP 257](https://peps.python.org/pep-0257/))
* Added an helper function, **get_valid_options**, to deal with parameters passed with each command who need it. format _'-abcd'_.
* Created a space **fn** to gather thoses common helper functions.
* Improved the **Manual.txt** file with errors and helper function to help for development while removed the command tail and head as it is now better documented in the man files and to avoid duplicate.

### CODE REVIEW

File: Manual.txt [hunk 1]
```diff
@@ -262,27 +251,51 @@ However it's highly recommended to have a root folder ("LjinuxRoot" or an sdcard
...
-
-based: Error
-    This is a general error
-    
-based: Filesystem unwritable, pi in developer mode
-    This error shows up when a write to the internal storage occures while developer mode is enabled. By CircuitPython design, when the internal storage is being made available from usb, it is unwritable to the os. To disable developer mode, delete the file "devm" from the root of LJINUX, and run the command "reboot"
-    
-based: Missing files
-    This error occurs when files needed for the command to run are missing.
+The differents errors used by the based shell.
```
Comment (bill88t) [inline Manual.txt:254]: typo, imma fix it myself

File: source/ljinux_source.py [hunk 2]
```diff
@@ -830,6 +830,12 @@ def get_bins():
                 return list()
         
         def error(wh=3, f=None):
+            """
+                The differents errors used by the based shell.
```
Comment (bill88t) [inline source/ljinux_source.py:834]: hmm, sus.

File: source/ljinux_source.py [hunk 3]
```diff
@@ -1497,32 +1505,66 @@ def catt(inpt): # kot
...
+                    lines = 10
+                    if inpt[1][0] == '-':
+                        ops = ljinux.based.fn.get_valid_options(inpt[1], 'n')
+                        if 'n' in ops and len(inpt) == 4:
+                            try:
+                                lines = int(inpt[2])
+                                file = inpt[3]
+                                del ops
+                            except IndexError:
+                                ljinux.based.error(9)
```
Comment (bill88t) [inline source/ljinux_source.py:1519]: need to add comments onto these

File: source/ljinux_source.py [hunk 4]
```diff
@@ -1497,32 +1505,66 @@ def catt(inpt): # kot
...
+                                    print(content[i], end="")
+                                else:
+                                    print(content[i])
+                            f.close()
+                            del content
+                            del count
+                            del lines
+                            del start
+                            del end
+                            return 1
```
Comment (bill88t) [inline source/ljinux_source.py:1555]: return 1??
Success = 0
Error = 1

**Review by bill88t — APPROVED** [review]
Looks good, imma put some extra comments & additions, test it, push it onto main tomorrow. It's 12 am, i'm out.

File: source/ljinux_source.py (hunk 2 above)
Comment (mdaadoun) [reply source/ljinux_source.py:834]: Useful comment hu..

**Comment by mdaadoun** [comment]
As for the Manual.txt, I removed the commands head/tail.

But maybe we could keep the list of all programs with a short description.

Separated by built-in/external.

Just for ref and keep usage for the man files.

**Comment by bill88t** [comment]
The Manual.txt is meant to a complete manual, containing everything you could possbly need. There is no need to keep it short.

**Pull request merged**